core/server/services/newsletters/service.js

```diff
@@ -112,10 +112,18 @@ class NewslettersService {
      * @public
      * @param {object} attrs model properties
      * @param {Object} [options] options
-     * @param {Object} [options] options.transacting
+     * @param {Object} [options] options.transacting Don't use this option outside of the service unless you also update the limit checking.
      * @returns {Promise<{object}>} Newsetter Model with verification metadata
      */
     async add(attrs, options = {}) {
+        // We need to make sure we always check the limits outside of transactions
+        // Or we'll get a deadlock in SQLite
+        if (!options.transacting) {
```

Comment:
IMO this is dangerous - we should have ways to skip verification.

For me it would be preferable to remove the `options.transaction` option, and usage of it.


We don't currently use it, and if we do in the future, I would much prefer it to not work/break than to bypass our limits!

Reply:
You are right! I'm going to see how I can make the option private or if it would be feasible to get the transaction over to the limit-service